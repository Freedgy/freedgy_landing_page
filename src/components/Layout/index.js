import React from 'react';
import PropTypes from 'prop-types';
import { MuiThemeProvider, createMuiTheme, withStyles } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';
import Header from '../Header';
import Footer from '../Footer';

let theme = createMuiTheme({

    typography: {
        useNextVariants: true,
        button: {
            '@media only screen and (max-width: 600px)': {
                fontWeight: 300,
                fontSize: "10px",
                lineHeight: "18px",
            },
            '@media only screen and (min-width: 600px)': {
                fontWeight: 300,
                fontSize: "12px",
                lineHeight: "20px",
            },
            '@media only screen and (min-width: 768px)': {
                fontWeight: 400,
                fontSize: "14px",
                lineHeight: "22px",
            },
            '@media only screen and (min-width: 992px)': {
                fontWeight: 400,
                fontSize: "14px",
                lineHeight: "22px",
            },
            '@media only screen and (min-width: 1200px)': {
                fontWeight: 500,
                fontSize: "16px",
                lineHeight: "24px",
            },
        },
        h5: {
            '@media only screen and (max-width: 600px)': {
                fontWeight: 500,
                fontSize: "16px",
                lineHeight: "20px",
            },
            '@media only screen and (min-width: 600px)': {
                fontWeight: 500,
                fontSize: "16px",
                lineHeight: "20px",
            },
            '@media only screen and (min-width: 768px)': {
                fontWeight: 600,
                fontSize: "18px",
                lineHeight: "22px",
            },
            '@media only screen and (min-width: 992px)': {
                fontWeight: 600,
                fontSize: "18px",
                lineHeight: "22px",
            },
            '@media only screen and (min-width: 1200px)': {
                fontWeight: 700,
                fontSize: "20px",
                lineHeight: "24px",
            },
        },
        h2: {
            '@media only screen and (max-width: 600px)': {
                lineHeight: "40px",
                fontWeight: 400,
                fontSize: "30px",
            },
            '@media only screen and (min-width: 600px)': {
                lineHeight: "40px",
                fontWeight: 400,
                fontSize: "30px",
            },
            '@media only screen and (min-width: 768px)': {
                lineHeight: "50px",
                fontWeight: 600,
                fontSize: "40px",
            },
            '@media only screen and (min-width: 992px)': {
                lineHeight: "50px",
                fontWeight: 600,
                fontSize: "40px",
            },
            '@media only screen and (min-width: 1200px)': {
                lineHeight: "60px",
                fontWeight: 700,
                fontSize: "50px",
            },
        },
        h1: {
            '@media only screen and (max-width: 600px)': {
                lineHeight: "48px",
                fontWeight: 500,
                fontSize: "40px",
            },
            '@media only screen and (min-width: 600px)': {
                lineHeight: "60px",
                fontWeight: 500,
                fontSize: "52px",
            },
            '@media only screen and (min-width: 768px)': {
                lineHeight: "72px",
                fontWeight: 600,
                fontSize: "64px",
            },
            '@media only screen and (min-width: 992px)': {
                lineHeight: "84px",
                fontWeight: 600,
                fontSize: "76px",
            },
            '@media only screen and (min-width: 1200px)': {
                lineHeight: "96px",
                fontWeight: 700,
                fontSize: "88px",
            },
        },
        body1: {
            '@media only screen and (max-width: 600px)': {
                fontWeight: 300,
                fontSize: "12px",
                lineHeight: "20px"
            },
            '@media only screen and (min-width: 600px)': {
                fontWeight: 300,
                fontSize: "12px",
                lineHeight: "20px"
            },
            '@media only screen and (min-width: 768px)': {
                fontWeight: 300,
                fontSize: "14px",
                lineHeight: "22px"
            },
            '@media only screen and (min-width: 992px)': {
                fontWeight: 300,
                fontSize: "14px",
                lineHeight: "22px"
            },
            '@media only screen and (min-width: 1200px)': {
                fontWeight: 400,
                fontSize: "16px",
                lineHeight: "24px"
            },
        },
        fontFamily: [
            'poppins',
            '-apple-system',
            'BlinkMacSystemFont',
            '"Segoe UI"',
            'Roboto',
            '"Helvetica Neue"',
            'Arial',
            'sans-serif',
            '"Apple Color Emoji"',
            '"Segoe UI Emoji"',
            '"Segoe UI Symbol"',
        ].join(','),
    },
    palette: {
        primary: {
            light: '#008080',
            main: '#008080',
            dark: '#008080',
        },
        secondary: {
            light: '#fff',
            main: '#fff',
            dark: '#fff',
        },
    },
    shape: {
        borderRadius: 8,
    },
});

const styles = {
    appContent: {
        "&&": {
            flex: 1,
            display: 'flex',
            flexDirection: 'column',
            backgroundColor: '#00a0a0'
        }
    },
    mainContent: {
        "&&": {
            flex: 1,
            padding: '0%',
        }
    },
};

function Layout({ location, children, classes, lang, datas }) {
    return (
        <MuiThemeProvider theme={theme}>
            <div className={classes.appContent}>
                <CssBaseline />
                <Header location={location} lang={lang} datas={datas.header} />
                <main className={classes.mainContent}>{children}</main>
                <Footer datas={datas.footer} />
            </div>
        </MuiThemeProvider>
    );
}

Layout.propTypes = {
    classes: PropTypes.object.isRequired,
    location: PropTypes.object.isRequired,
    children: PropTypes.object,
};

export default withStyles(styles)(Layout);
