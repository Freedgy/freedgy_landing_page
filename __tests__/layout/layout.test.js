import React from "react"
import renderer from "react-test-renderer"

import Layout from "../../src/components/Layout"

import jsdom from 'jsdom';
const {JSDOM} = jsdom;
const {document} = (new JSDOM('<!doctype html><html><body></body></html>')).window;
global.document = document;
global.window = document.defaultView;
global.Image = window.Image;

describe("Layout", () => {
    it("renders correctly", async () => {
        const datas = require("../../static/pages/pages.en.json");

        const tree = renderer
            .create(<Layout lang="en" datas={datas.layout} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})