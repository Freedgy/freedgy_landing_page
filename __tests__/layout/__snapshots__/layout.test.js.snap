// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="Layout-appContent-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary Header-appBarTransparent-4 mui-fixed MuiPaper-elevation0"
  >
    <div
      className="MuiBox-root MuiBox-root-10 Header-container-6"
    >
      <div
        className="MuiBox-root MuiBox-root-11"
      >
        <img
          alt="watermark_freedgy"
          className="Header-logo-3"
          src="/images/logo/watermark.png"
        />
      </div>
      <div
        className="MuiBox-root MuiBox-root-12 Header-navBar-7"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-8"
          href="#home"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            home
          </span>
        </a>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-8"
          href="#about"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            about
          </span>
        </a>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-8"
          href="#team"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            team
          </span>
        </a>
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-text Header-navItemTransparent-8"
          href="#contact"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            contact
          </span>
        </a>
      </div>
      <div
        className="MuiBox-root MuiBox-root-13"
      >
        <div
          className="ReactFlagsSelect-module_flagsSelect__2pfa2"
          data-testid="rfs"
        >
          <button
            aria-expanded={false}
            aria-haspopup="listbox"
            aria-labelledby="rfs-btn"
            className="ReactFlagsSelect-module_selectBtn__19wW7"
            data-testid="rfs-btn"
            disabled={false}
            id="rfs-btn"
            onClick={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "fontSize": 16,
              }
            }
            type="button"
          >
            <span
              className="ReactFlagsSelect-module_selectValue__152eS"
            >
              Select a country
            </span>
          </button>
        </div>
      </div>
    </div>
  </header>
  <main
    className="Layout-mainContent-2"
  />
  <div
    className="MuiBox-root MuiBox-root-15 Footer-footer-14"
  >
    <div
      className="MuiBox-root MuiBox-root-16"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter"
      >
        Copyright Â© 
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="https://freedgy.com/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Freedgy
        </a>
         
        2021
        .
      </p>
    </div>
    <div
      className="MuiBox-root MuiBox-root-17"
    >
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        href="https://www.instagram.com/freedgy.project"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          <svg
            color="#008080"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#008080",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
            />
          </svg>
        </span>
      </a>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        href="https://www.linkedin.com/company/freedgy/"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          <svg
            color="#008080"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "#008080",
              }
            }
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
</div>
`;
